src/modules/m_spanningtree/netburst.cpp

```diff
@@ -140,25 +140,30 @@ void TreeSocket::SendFJoins(Channel* c)
 		buffer.append(list).append("\r\n");
 	}
 
-	int linesize = 1;
+	int linesize = 0;
 	for (BanList::iterator b = c->bans.begin(); b != c->bans.end(); b++)
 	{
-		int size = b->data.length() + 2;
-		int currsize = linesize + size;
-		if (currsize <= 350)
-		{
-			modes.append("b");
-			params.append(" ").append(b->data);
-			linesize += size;
-		}
-		if ((modes.length() >= ServerInstance->Config->Limits.MaxModes) || (currsize > 350))
+		int size = b->data.length() + 2; /* "b" and " " */
+		int nextsize = linesize + size;
+
+		if ((modes.length() >= ServerInstance->Config->Limits.MaxModes) || (nextsize > FMODE_MAX_LENGTH))
 		{
-			/* Wrap at MAXMODES */
-			buffer.append(":").append(ServerInstance->Config->GetSID()).append(" FMODE ").append(c->name).append(" ").append(ConvToStr(c->age)).append(" +").append(modes).append(params).append("\r\n");
+			/* Wrap */
+			buffer.append(":")
+			      .append(ServerInstance->Config->GetSID())
+			      .append(" FMODE ")
+			      .append(c->name).append(" ")
+			      .append(ConvToStr(c->age))
+			      .append(" +").append(modes).append(params).append("\r\n");
```

Comment:
I don't see the meaning of the long line being changed, it seems to be the same except broken up. Please avoid doing this to minimize the diff.